event/src/unvalidated/event.rs

```diff
@@ -18,7 +18,7 @@ pub enum Event<D> {
 
 /// Ceramic Event as it is encoded in the protocol.
 #[derive(Serialize, Deserialize)]
-#[serde(untagged)]
+#[serde(untagged, deny_unknown_fields)]
```

Comment:
I'm not convinced we want to be this strict. I'm trying to articulate/think through it so bear with me. Some of it is probably that the events/schemas aren't self describing. 

I think doing this will basically make it impossible to be an old node on the network, as no one will be able to talk to you if we ever added a new field (you'll reject every event they send you).

I think it may make sense to have 2 versions (one for API ingest and one for sync). We could reject anything coming in on the API outright, but we shouldn't do that to peers if they are running a newer version of the protocol than we are. It feels like this makes everything a hard fork. I'd be happy to be wrong about this so shoot me down 😄.

Reply:
Agreed this is a problem, going to need a different approach